src/main/scala/org/broadinstitute/dsde/firecloud/service/AttributeSupport.scala

```diff
@@ -0,0 +1,34 @@
+package org.broadinstitute.dsde.firecloud.service
+
+import org.broadinstitute.dsde.firecloud.model.Attributable._
+import org.broadinstitute.dsde.firecloud.model.AttributeUpdateOperations._
+import org.broadinstitute.dsde.firecloud.model._
+
+/**
+  * Created by putnam on 11/1/16.
+  */
+trait AttributeSupport {
+  /**
+    * given a set of existing attributes and a set of new attributes, calculate the attribute operations
+    * that need to be performed
+    */
+  def generateAttributeOperations(existingAttrs: AttributeMap, newAttrs: AttributeMap, attributeFilter: AttributeName => Boolean): Seq[AttributeUpdateOperation] = {
+    val oldKeys = existingAttrs.keySet.filter(attributeFilter)
+    val newFields = newAttrs.seq.filterKeys(attributeFilter)
```

Comment:
`filterKeys` is unsafe, and we should avoid it.

See https://issues.scala-lang.org/browse/SI-4776 and broadinstitute/rawls#390 ... in that other PR we spent a long time tracking down mysterious errors; let's avoid those here.

One possible solution: rewrite so that both old and new operate on the entire map, instead of using keys for old and map for new? That way, the `attributeFilter` argument can operate against `AttributeMap.filter`.